src/App.tsx

```diff
@@ -1,26 +1,195 @@
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { deleteTodos, getTodos, updateTodos, USER_ID } from './api/todos';
+import { Footer } from './components/Footer/Footer';
+import { Status } from './types/Status';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [statusOfTodos, setStatusOfTodos] = useState(Status.all);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingIds, setloadingIds] = useState<number[]>([]);
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+  //const [isEditing, setIsEdiding] = useState(false);
+
+  function loadTodos() {
+    setIsLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }
+
+  if (errorMessage) {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  let filteredTodos = todos;
+
+  if (statusOfTodos === Status.active) {
+    filteredTodos = filteredTodos.filter(todo => !todo.completed);
+  }
+
+  if (statusOfTodos === Status.completed) {
+    filteredTodos = filteredTodos.filter(todo => todo.completed);
+  }
+
+  async function handleDeleteTodo(id: number) {
+    setloadingIds(prev => [...prev, id]);
+    setDeletingTodoId(id);
+
+    try {
+      await deleteTodos(id);
+      setTodos(currentTodo => currentTodo.filter(todo => todo.id !== id));
+    } catch (error) {
+      setErrorMessage('Unable to delete a todo');
+    } finally {
+      setDeletingTodoId(null);
+      setloadingIds(prev => prev.filter(todoId => todoId !== id));
+    }
+  }
+
+  function deleteCompletedTodo() {
+    const completedTodo = todos.filter(todo => todo.completed);
+
+    completedTodo.forEach(todo => handleDeleteTodo(todo.id));
+  }
+
+  const handleUpdateTodo = ({ id, title, completed }: Todo) => {
+    const todoToUpdate = todos.find(todo => todo.id === id);
+
+    if (!todoToUpdate) {
+      return Promise.reject(new Error('Todo not found'));
+    }
+
+    const trimmedTitle = title.trim();
+    const updatedTodo = {
+      ...todoToUpdate,
+      title: trimmedTitle,
+      completed: completed ?? todoToUpdate.completed,
+    };
+
+    setloadingIds(currentId => [...currentId, id]);
+
+    return updateTodos(id, updatedTodo)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.map(todo => (todo.id === id ? updatedTodo : todo)),
+        );
+      })
```

Comment:
```suggestion
  const handleUpdateTodo = ({ id, title, completed }: Todo, onSuccess?: VoidFunction) => {
    const todoToUpdate = todos.find(todo => todo.id === id);

    if (!todoToUpdate) {
      return Promise.reject(new Error('Todo not found'));
    }

    const trimmedTitle = title.trim();
    const updatedTodo = {
      ...todoToUpdate,
      title: trimmedTitle,
      completed: completed ?? todoToUpdate.completed,
    };

    setloadingIds(currentId => [...currentId, id]);

    return updateTodos(id, updatedTodo)
      .then(() => {
        setTodos(currentTodos =>
          currentTodos.map(todo => (todo.id === id ? updatedTodo : todo)),
        );
        
        onSuccess?.();
      })
```

and use onSuccess handler, which you passed from todoItem component here.